http-clients/src/main/java/org/triplea/http/client/web/socket/GenericWebSocketClient.java

```diff
@@ -68,7 +71,10 @@ public void send(final OutgoingT message) {
 
   /** Non-blocking close of the websocket connection. */
   public void close() {
-    client.close();
+    CompletableFutureUtils.logExceptionWhenComplete(
+        CompletableFuture.runAsync(client::close, threadPool),
+        e -> log.log(Level.WARNING, "Failed to close client", e));
```

Comment:
Do you remember if warnings are shown (as a pop-up) to users by default?

Reply:
`ErrorMessageHandler` defines `Level.WARNING` as minimum level

Reply:
Thanks for checking, we may want to consider the error handling; errors tend to have users overreact and given we're dealing with a thick client, logging is of very limited value.

Perhaps if the connection winds up being closed, we can `info` or `debug` log this so there is no pop-up, effectively handling the error. If the connection is still open, we could consider a retry, though I'm not sure if that is really worth it.

Reply:
I'll leave it up to you to decide this, the log statements are all in one place now, so it's easy to change them all.

Reply:
I think the consideration is more or less as follows:
- are we going to be spamming users with error notifications? For example, will this be a N+1 error notification, will there be some other error beforehand and then a user sees this error message popup.
- Do we want this error reported? A popup will have a 'report to triplea' attached to it, is this something that we will want users to report to us?
- Do we want to expect the user to restart TripleA after seeing this? For better or worse, most users assume on any error that they should restart TripleA. I've observed that behavior often enough I consider it a rule of thumb, it's a bit funny sometimes when errors are relatively benign, but people go through the trouble of rehosting a game and restarting everything.